source/agora/common/Block.d

```diff
@@ -55,3 +56,51 @@ public struct Block
     ///
     public Transaction tx;
 }
+
+/*******************************************************************************
+
+    Create a new block, referencing the provided previous block.
+
+    Params:
+        prev_block = the previous block
+        tx = the transaction that will be contained in the new block
+
+    In the current preliminary design a block contains a single transaction.
+
+*******************************************************************************/
+
+public Block makeNewBlock ( Block prev_block, Transaction tx )
+{
+    Block block;
+
+    block.header.prev_block = hash(prev_block);
+    block.header.height = prev_block.header.height + 1;
+    block.header.tx_hash = hash(tx);
+    block.tx = tx;
+
+    return block;
+}
+
+///
+unittest
+{
+    Block gen_block = getGenesisBlock();
+    auto new_block = makeNewBlock(gen_block, Transaction.init);
+    assert(new_block.header.prev_block == hash(gen_block));
+}
+
+/*******************************************************************************
+
+    Returns:
+        the genesis block
+
+*******************************************************************************/
+
+public Block getGenesisBlock ( )
+{
+    Block block;
+
+    block.header.height = 0;  // genesis block is 0
```

Comment:
That's the default?

Reply:
However, there should be a transaction giving 40k to a `random()` keypair.

Reply:
Makes sense.

Reply:
Speaking of which, we need to define the unitary model or just a helper function (so 1 BOA = 1,000,000 units or whatever we defined it as). So I could do something like `output.amount = 1.boa` and `boa` is defined as `long boa (long amount) { return amount * 1_000_000; }`.

That could be a separate task in the backlog.

Reply:
Correct